bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -731,7 +735,8 @@ void DWARFRewriter::updateDebugInfo() {
   auto TheTriple = std::make_unique<Triple>(File->makeTriple());
   std::unique_ptr<DIEStreamer> Streamer =
       createDIEStreamer(*TheTriple, *ObjOS, "TypeStreamer", DIEBlder, *this);
```

Comment:
Can we pass GDBIndexSection into createDIEStreamer like we do for DIEBuilder so we dont't have to pass it through finalizeTypeSections --> emitUnit?

Reply:
Wouldn't we still have to pass it through emitUnit? The flow is finalizeTypeSections -> emitUnit -> DIEStreamer::emitUnit -> emitUnitHeader -> emitTypeUnitHeader where GDBIndex::addGDBTypeUnitEntry is called. emitTypeUnitHeader is private so we wouldn't be able to call it from createDIEStreamer.

Reply:
createDIEStreamer creates DIEStreamer which takes/stores DIEBulldr, etc. So same way pass in GDBIndexSection, so then in emitTypeUnitHeader we can call it like we do DIEBldr